atomic_reactor/utils/operator.py

```diff
@@ -734,6 +734,30 @@ def _replace_pullspecs_not_in_annotations(self, obj, k_or_i, replacements):
             # in replacement_pullspecs and no replacement will be done
             self._replace_unnamed_pullspec(obj, k_or_i, replacements)
 
+    def modifications_append(self, append_mods):
+        if not append_mods:
+            return
+        for mod in append_mods:
+            for key in append_mods[mod]:
+                item = append_mods[mod][key]
+                if isinstance(item, list):
+                    self.data[mod].setdefault(key, []).extend(item)
+                else:
+                    #  Raise something - we're expecting a list
+                    pass
+        check_csv(self.data, _mini_csv_schema)
+
+    def modifications_update(self, update_mods):
+        if not update_mods:
+            return
+        for mod in update_mods:
+            for key in update_mods[mod]:
+                if isinstance(self.data[mod].get(key), dict):
```

Comment:
If I understand correctly, this code works in a following way
```
>>> data = { 'a': {'b': {'c': {'something': 1}}}}                                                                                                                                                                                          
>>> update_spec = { 'a': {'b': {'c': {'else': 9}}}}                                                                                                                                                                                        
>>> data['a']['b'].update(update_spec['a']['b'])                                                                                                                                                                                           
>>> print(data)                                                                                                                                                                                                                                   
{'a': {'b': {'c': {'else': 9}}}}
```

However I'd expect: 
```
{'a': {'b': {'c': {'else': 9, 'something': 1}}}}
```

Reply:
Ah. It needs to be recursive.